include/cadmium/engine/pdevs_simulator.hpp

```diff
@@ -216,20 +163,124 @@ namespace cadmium {
+                LOGGER::template log<cadmium::logger::logger_state, decltype(formatter::log_state), const typename model_type::state_type&>(formatter::log_state, _model.state);
             }
     //TODO: use enable_if functions to give access to read state and messages in debug mode
         };
 
+
+        template<template<typename T> class MODEL, typename TIME, typename LOGGER>
+        class dynamic_simulator {
+            using formatter=typename cadmium::logger::simulator_formatter<MODEL, TIME>;
+
+            using input_ports=typename modeling::dynamic_atomic<MODEL, TIME>::input_ports;
+            using output_ports=typename modeling::dynamic_atomic<MODEL, TIME>::output_ports;
+            using in_bags_type=typename make_message_bags<input_ports>::type;
+            using out_bags_type=typename make_message_bags<output_ports>::type;
+            modeling::dynamic_atomic<MODEL, TIME> _model;
+            TIME _last;
+            TIME _next;
+
+            dynamic_message_bags _outbox;
+            dynamic_message_bags _inbox;
+
+        public:
+            using model_type=MODEL<TIME>;
+
+            /**
+             * @brief constructor is used as init function, sets the start time
+             * @param initial_time is the start time
+             */
+            void init(TIME initial_time) {
+                LOGGER::template log<cadmium::logger::logger_info, decltype(formatter::log_info_init), TIME>(formatter::log_info_init, initial_time);
+
+                _last=initial_time;
+                cadmium::concept::atomic_model_assert<MODEL>();
+                _next = initial_time + _model.time_advance();
+
+                LOGGER::template log<cadmium::logger::logger_state, decltype(formatter::log_state), const typename model_type::state_type&>(formatter::log_state, _model.state);
+            }
+
+            TIME next() const noexcept{
+                return _next;
+            }
+
+            void collect_outputs(const TIME &t) {
+                LOGGER::template log<cadmium::logger::logger_info, decltype(formatter::log_info_collect), TIME>(formatter::log_info_collect, t);
+
+                //cleanning the inbox and producing outbox
+                _inbox = dynamic_message_bags();
+
+                if (_next < t){
+                    throw std::domain_error("Trying to obtain output when not internal event is scheduled");
+                } else if (_next == t) {
+                    _outbox = _model.output();
+                } else {
+                    _outbox = dynamic_message_bags();
+                }
+
+                LOGGER::template log<cadmium::logger::logger_messages, decltype(formatter::log_messages_collect), out_bags_type>(formatter::log_messages_collect, _outbox);
+            }
+
+            /**
+             * @brief outbox keeps the output generated by the last call to collect_outputs
+             */
+            dynamic_message_bags outbox() const noexcept{
```

Comment:
Using the stack, these methods may not be easy to define as noexcept anymore. Maybe we should remove it.

Reply:
I think there is no problem because is a constant expression returning a valid object, if there is a stack problem with it, then the problem should cause an exception before when assigning the value to the attribute. But I've follow your suggestion and removed the noexcept keyword.